### DISCUSSION TITLE
Sea ice velocity does not converge? -- Truncated too many times

### DISCUSSION THREAD

**Discussion by zi-hua-liu** [NOAA-GFDL/CEFI-regional-MOM6#26] [Q&A] [answered]
I am running the regional MOM6 model in MOM6-SIS2 coupled mode. However, the model has encountered two fatal errors that halted its execution. I have provided details of [my model setup](https://drive.google.com/file/d/1K_5TwENnX1brabIj-FvhIjdP-_9HWavl/view?usp=sharing) here in case it aids in diagnosing the issue.

- The model crashed with an error message indicating 'Sea ice velocity has been truncated too many times' (can seen from 'mom6_***.err' file). I am uncertain whether this error stems from any misconfigured parameters in my SIS_input or MOM_input setup. These parameters are consistent with those used in our global MOM6 model, which runs without any issues. ---- Just noticed that this issue might come from my boundary conditions, where input data in two boundaries (with numbers *002*, *004*) are wrong (constant values everywhere).
- In the MOM_override file, if the setup related to "tideamp" is activated, the model crashes during the initialization stage. I suspect that there might be an issue with the data type or format of the "tideamp" input, but I am not sure about the specific problem. Are there any template files available that address the input data for "tideamp"?

Any comments are appreciated! Thanks!

**Comment by theresa-cordero** [comment]
The truncation error is from the sea-ice velocity exceeding the limit set by the CFL criterion. If you regional model has a higher resolution than your global model you may need to decrease your time steps in the ocean and sea-ice model to maintain stability. But it sounds like you may have figured out the source of the of the incorrect velocities! I recommend testing your domain without OBC's when you are first setting it up.

For the tidal amplitude, I think you need to change the name of the variable in your tideamp_regional_nep.nc file to just "tideamp" rather than "tideamp_interp". I don't think the code allows you to specify the name of that field.

**Reply by zi-hua-liu** [reply]
Thank you Theresa for the comments and suggestions! I attempted to rename the variable to 'tideamp,' yet the error persists. So it might not be the name issue?

**Reply by theresa-cordero** [reply]
I am looking at the OM4_025 case and the Arctic 12km simulation as an example of the tidal amplitude file. 

One key difference is that your lon/lat variables are 2D, where the Arctic and OM4_025 use a 1 dimensional x and y coordinate. Your lon/lat look uniform to me, so hopefully it will be easy to replace the 2D fields with 1D. 

Could you post the error message you get about the tidal amplitude file ?

**Reply by zi-hua-liu** [reply]
Thank you Theresa for the support and quick response. The error message was "FATAL from PE  2: open_netcdf_file: File INPUT/Invalid argument". 

I checked the input file from our global mom6 simulations, it shows that 'geothermal' input file uses 1 dimensional i and j index, but 'tideamp' file use 2 dimensional lat and lon coordinates. I think I have also tried to use 1 dimensional i and j index in 'tideamp' but got similar error, so it might not be using i and j as in 'geothermal'. I will replace the 2D lat and lon coordinates to 1D lat and lon. Will let you know if that works soon. Thanks!

**Reply by theresa-cordero** [reply]
That error makes me think the error might be with the file name you are giving. 
You are updating your MOM_override file to have:
```
#override INT_TIDE_DISSIPATION = True 
#override READ_TIDEAMP = True
#override TIDEAMP_FILE = "tideamp_regional_nep.nc"
```
when using this option, correct?

**Reply by zi-hua-liu** [reply]
Yes, that is correct. I updated my MOM_override with those above three functions to include 'tideamp'.

**Reply by zi-hua-liu** [reply]
May I ask for an 'tideamp' data example from OM4_025 or Arctic 12km cases? I have tried to set x/y to 1D instead of using 2D lon/lat, but it still does not work, with the same error... So I was thinking to reset my 'tideamp' data to be the same type as in the examples as you mentioned.

**Reply by yichengt900** [reply, marked as answer]
Hi @zi-hua-liu,

After reviewing your configuration, I believe the issue you encountered is due to the absence of `H2_FILE` (Variance of sub-grid scale topography) in your model configuration. This information can be found in the original global INPUT `ocean_topog.nc`. Below is an example script that can be used to regrid H2 data:

```
import numpy as np
import xarray
import xesmf

import sys

#First load regional grid
hgrid = xarray.open_dataset('ocean_geometry.nc')

mom_grid = {
    'lon': hgrid.lonh,
    'lat': hgrid.lath,
}


tide = xarray.open_dataset(
    '/collab1/data_untrusted/Xiao.Liu/Iron_Project/WHOI_NEP/INPUT/INPUT/tidal_amplitude_TPXO9_OM4p125.v20200922.nc',
)

h2 = xarray.open_dataset(
    '/collab1/data_untrusted/Xiao.Liu/Iron_Project/WHOI_NEP/INPUT/INPUT/ocean_topog.nc',
)

tide['h2'] = h2['h2']

tide_grid = {
    'lon': tide.lon,
    'lat': tide.lat,
}

tide_to_mom = xesmf.Regridder(tide_grid, mom_grid, method='bilinear', reuse_weights=False)
regridded = tide_to_mom(tide)

print(regridded)

# Rename variables
ds = regridded.rename_vars({'lath': 'ny', 'lonh': 'nx'})

ds.to_netcdf(
    './tideamp.nc', format='netcdf4' 
)
```

You also have to add the following lines in your MOM_override:
```
#override INT_TIDE_DISSIPATION = True 
#override READ_TIDEAMP = True
#override TIDEAMP_FILE = "tideamp.nc"
#override TIDEAMP_VARNAME = "tideamp"
#override H2_FILE = "tideamp.nc"
```

Keep in mind we normally simulate tide explicitly in regional application (e.g. `OBC_TIDE_CONSTITUENTS = "M2,S2,N2,K2,K1,O1,P1,Q1,MM,MF"`).

**Reply by zi-hua-liu** [reply]
Thank you very much! It works after adding H2 data. 

Later on, I may explicitly simulate tides in the regional model to compare the differences with the existing vertical tidal mixing parameterization. I appreciate your suggestions.
